plugins/packages/appwrite/lib/index.ts

```diff
@@ -26,22 +32,36 @@ export default class Appwrite implements QueryService {
           );
           break;
         case 'get_document':
-          result = await getDocument(database, queryOptions.collectionId, queryOptions.documentId);
+          if (!queryOptions.documentId) throw new Error('Document id is required');
+          result = await getDocument(database, database_id, queryOptions.collectionId, queryOptions.documentId);
           break;
         case 'add_document':
-          result = await createDocument(database, queryOptions.collectionId, body);
+          if (isBodyEmpty) throw new Error('Body is required');
+          result = await createDocument(database, database_id, queryOptions.collectionId, body as object);
           break;
         case 'update_document':
-          result = await updateDocument(database, queryOptions.collectionId, queryOptions.documentId, body);
+          if (!queryOptions.documentId) throw new Error('Document id is required');
+          if (isBodyEmpty) throw new Error('Body is required');
+          result = await updateDocument(
+            database,
+            database_id,
+            queryOptions.collectionId,
+            queryOptions.documentId,
+            body as object
+          );
           break;
         case 'delete_document':
-          result = await deleteDocument(database, queryOptions.collectionId, queryOptions.documentId);
+          if (!queryOptions.documentId) throw new Error('Document id is required');
+          result = await deleteDocument(database, database_id, queryOptions.collectionId, queryOptions.documentId);
           break;
         case 'bulk_update':
+          if (!queryOptions.records) throw new Error('Records field is required');
+          if (!queryOptions['document_id_key']) throw new Error('Key for document id field is required');
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/ToolJet/ToolJet/pull/10024#issuecomment-2162145568>#befc40</a><span> - 06/12/2024, 05:43 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The error handling for missing 'document_id_key' is inconsistent with the rest of the code. It throws a generic 'Error' instead of a 'QueryError', which is used elsewhere in the code for consistency.
</div>
<div id="fix">
<b>Fix:</b> Replace the generic 'Error' with a 'QueryError' to maintain consistency in error handling.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div>
<pre><code>
 @@ -58,7 +58,7 @@ export default class Appwrite implements QueryService {
           if (!queryOptions.records) throw new Error('Records field is required');
 -          if (!queryOptions['document_id_key']) throw new Error('Key for document id field is required');
 +          if (!queryOptions['document_id_key']) throw new QueryError('Key for document id field is required', 'Missing document_id_key', {});
           result = await bulkUpdate(
             database,
             database_id,
</code></pre>
</div>
</details>
</div>
<br>


</div>